core/ArchiveProcessor/RecordBuilder.php

```diff
@@ -0,0 +1,241 @@
+     * @var int
+     */
+    protected $maxRowsInSubtable;
+
+    /**
+     * @var string|null
+     */
+    protected $columnToSortByBeforeTruncation;
+
+    /**
+     * @var int
+     */
+    protected $blobReportLimit;
+
+    /**
+     * @var array|null
+     */
+    protected $columnAggregationOps;
+
+    /**
+     * @param int|null $maxRowsInTable
+     * @param int|null $maxRowsInSubtable
+     * @param string|null $columnToSortByBeforeTruncation
+     * @param array|null $columnAggregationOps
+     */
+    public function __construct(?int $maxRowsInTable = null, ?int $maxRowsInSubtable = null,
+                                ?string $columnToSortByBeforeTruncation = null, ?array $columnAggregationOps = null)
+    {
+        $this->maxRowsInTable = $maxRowsInTable;
+        $this->maxRowsInSubtable = $maxRowsInSubtable;
+        $this->columnToSortByBeforeTruncation = $columnToSortByBeforeTruncation;
+        $this->columnAggregationOps = $columnAggregationOps;
+    }
+
+    public function isEnabled(ArchiveProcessor $archiveProcessor): bool
+    {
+        return true;
+    }
+
+    public function build(ArchiveProcessor $archiveProcessor): void
+    {
+        if (!$this->isEnabled($archiveProcessor)) {
+            return;
+        }
+
+        $recordsBuilt = $this->getRecordMetadata($archiveProcessor);
+
+        $recordMetadataByName = [];
+        foreach ($recordsBuilt as $recordMetadata) {
+            $recordMetadataByName[$recordMetadata->getName()] = $recordMetadata;
+        }
+
+        $numericRecords = [];
+
+        $records = $this->aggregate($archiveProcessor);
+        foreach ($records as $recordName => $recordValue) {
+            if (empty($recordMetadataByName[$recordName])) {
+                if ($recordValue instanceof DataTable) {
+                    Common::destroy($recordValue);
+                }
+                continue;
+            }
+
+            if ($recordValue instanceof DataTable) {
+                $record = $recordMetadataByName[$recordName];
+
+                $maxRowsInTable = $record->getMaxRowsInTable() ?? $this->maxRowsInTable;
+                $maxRowsInSubtable = $record->getMaxRowsInSubtable() ?? $this->maxRowsInSubtable;
+                $columnToSortByBeforeTruncation = $record->getColumnToSortByBeforeTruncation() ?? $this->columnToSortByBeforeTruncation;
+
+                $this->insertRecord($archiveProcessor, $recordName, $recordValue, $maxRowsInTable, $maxRowsInSubtable, $columnToSortByBeforeTruncation);
+
+                Common::destroy($recordValue);
+            } else {
+                // collect numeric records so we can insert them all at once
+                $numericRecords[$recordName] = $recordValue;
+            }
+        }
+        unset($records);
+
+        if (!empty($numericRecords)) {
+            $archiveProcessor->insertNumericRecords($numericRecords);
+        }
+    }
+
+    public function buildMultiplePeriod(ArchiveProcessor $archiveProcessor): void
```

Comment:
Can we add a docblock explaining in simple terms what the method does and how does it differ from `build` method above?

Does it build for multiple periods? Or build multiple records for a period? Or something else? Can we think of a better name perhaps?